agents/web-api/src/main/java/edu/jhuapl/dorset/agents/StockAgent.java

```diff
@@ -0,0 +1,277 @@
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            logger.debug("Failed to load stockagent/NASDAQ_Companies.csv");
+        }
+
+        InputStream nyseCompanies = StockAgent.class.getClassLoader()
+                .getResourceAsStream("stockagent/NYSE_Companies.csv");
+
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader(new InputStreamReader(nyseCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+
+                if (!this.stockSymbolMap.containsKey(companyInfo.getName())) {
+                    this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+                }
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            logger.debug("Failed to load stockagent/NYSE_Companies.csv");
+        } 
+
+    }
+
+    @Override
+    public AgentResponse process(AgentRequest request) {
+        logger.debug("Handling the request: " + request.getText());
+
+        String requestCompanyName = request.getText();
+        String[] stockCompanyInfo = findStockSymbol(requestCompanyName);
+        String keywordCompanyName = null;
+        String keywordCompanySymbol = null;
+
+        if (stockCompanyInfo[0] != null) {
+            keywordCompanyName = stockCompanyInfo[0].substring(0, 1).toUpperCase()
+                    + stockCompanyInfo[0].substring(1, stockCompanyInfo[0].length());
+            keywordCompanySymbol = stockCompanyInfo[1];
+        }
+
+        String json = null;
+
+        if (keywordCompanySymbol == null) {
+            return new AgentResponse(
+                    "I am sorry, I don't understand which company you are asking about.");
+        } else {
+            json = requestData(keywordCompanySymbol);
+
+        }
+
+        if (json == null) {
+            if (keywordCompanyName != null) {
+                return new AgentResponse(("I am sorry, I can't find the proper stock data for the company "
+                                + keywordCompanyName + ".").replace("..", "."));
+            } else {
+                return new AgentResponse(ResponseStatus.Code.AGENT_DID_NOT_KNOW_ANSWER);
+            }
+
+        } else {
+            JsonObject returnObj = processData(json, keywordCompanyName);
+
+            if (returnObj != null) {
+                return new AgentResponse(Response.Type.JSON,
+                        ("Here is the longitudinal stock market data from the last "
+                                + DAYS_IN_A_MONTH
+                                + " days for "
+                                + keywordCompanyName + ".").replace("..", "."),
```

Comment:
Added comment "// replace ".." with "." to maintain proper grammar when the keyword contains an abbreviation"